battle-engine.js

```diff
@@ -125,7 +125,13 @@ BattlePokemon = (() => {
 			this.battle.debug('Unidentified species: ' + this.species);
 			this.baseTemplate = this.battle.getTemplate('Unown');
 		}
-		this.species = this.baseTemplate.species;
+		if (this.baseTemplate.otherForms && this.baseTemplate.otherForms.indexOf(set.species) >= 0 && this.baseTemplate.baseSpecies !== 'Unown') {
+			let form = set.species.slice(this.baseTemplate.species.length);
+			this.form = form[0].toUpperCase() + form.slice(1);
+			this.species = this.baseTemplate.species + '-' + this.form;
+		} else {
+			this.species = this.baseTemplate.species;
+		}
```

Comment:
Specifically, what's wrong with doing

    this.species = set.species;

Seems a lot easier than your 7-liner mess?